src/gstreamer_bridge.cpp

```diff
@@ -1,22 +1,58 @@
 #include "gstreamer_bridge.hpp"
 
-GstreamerRosBridge::GstreamerRosBridge(ros::NodeHandle &nh)
-{
-    std::string camera_location = "/dev/video0";
-    nh.getParam("camera_location", camera_location);
-    int camera_width, camera_height, camera_fps;
-    nh.getParam("camera_width", camera_width);
-    nh.getParam("camera_height", camera_height);
-    nh.getParam("camera_fps", camera_fps);
-
-    cap_ = cv::VideoCapture("/dev/video0", cv::CAP_V4L2);
-    cap_.set(cv::CAP_PROP_FRAME_WIDTH, camera_width);  // Set frame width
-    cap_.set(cv::CAP_PROP_FRAME_HEIGHT, camera_height); // Set frame height
-    cap_.set(cv::CAP_PROP_FPS, camera_fps);            // Set frame rate
+// function to replace opencv resize since it seg faults in version 4.2 
+cv::Mat resizeAndPad(cv::Mat &image, int target_width, int target_height) {
```

Comment:
Might be just me worrying about premature optimization, but I'm concerned about the performance overhead introduced by this, especially on a raspberry pi. It's always run even if the target is the same size, and it does a pixel-wise copy. I suggest we add a check, and if no resizing is necessary then this should be a no-op. My assumption is that (at least for SAE) we can probably try to run everything at a camera-native resolution so we can avoid overhead from scaling.

Reply:
Sure a check can be added. I just added this in case 720p is too much for the network to handle. I can also run a CPU test to see if util is lower when it's commented out. It should be almost similar speeds to OpenCV's resize linear interpolation algorithm minus all of their hardware optimizations

Reply:
Right, yeah I forgot this is sent over the network, so it's not practical to use the same resolution for CV and streaming. For some reason I thought we could get away with it.

You mentioned that OpenCV resize segfaults here? Is there no workaround except making our own? Or what if we get gstreamer to do it for us, like with [this SO question](https://stackoverflow.com/questions/71955148/straightforward-way-to-downscale-too-large-video-with-gstreamer) it seems that if you specify the width and height twice in the pipeline, with a `videoscale` in the middle, it will do the scaling for you. The code here works but as you mentioned, I'm not sure if hardware optimizations will make a difference, e.g. using SIMD instructions or something.